src/openvpn/ssl_openssl.c

```diff
@@ -2104,6 +2104,13 @@ print_cert_details(X509 *cert, char *buf, size_t buflen)
         {
             type = "unknown type";
         }
+#else /* OpenSSL >= 3 */
+        type = EVP_PKEY_get0_type_name(pkey);
+        if (type == NULL)
+        {
+            type = "(error getting OpenSSL3 type)";
```

Comment:
If I saw that as a user, I would not understand it at all.  May I suggest something like `"(error getting public key type)"`?